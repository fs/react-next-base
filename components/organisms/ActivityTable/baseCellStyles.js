import { css } from 'styled-components';

const baseCellStyles = ({ up, breakpoints }) => css`
  border-bottom: 1px solid #e1e1e1;
  padding: 0.8rem 0.6rem;
  font-size: 0.8rem;
  line-height: 1.6;

  ${up(breakpoints.lg)} {
    padding: 1.2rem 1rem;
    font-size: 0.9rem;
  }

  &:first-child {
    padding-left: 0;
  }
`;

export default baseCellStyles;
