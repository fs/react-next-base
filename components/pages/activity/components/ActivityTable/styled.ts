import styled, { css } from 'styled-components';
import baseCellStyles from './baseCellStyles';

export const StyledTable = styled.table(
  ({ theme: { colors, up, breakpoints } }) =>
    css`
      position: relative;
      border-spacing: 0;
      display: block;
      overflow-x: auto;
      text-align: left;
      width: 100%;
      color: ${colors.darkGrey};

      ${up(breakpoints.lg)} {
        display: table;
        overflow-x: initial;
      }
    `,
);

export const ColorLabel = styled.td(
  ({ color }) => css`
    min-width: 5px;
    background-color: ${color};
  `,
);

export const HeaderCell = styled.th(
  ({ theme }) => css`
    ${baseCellStyles(theme)}
  `,
);

export const UserInfo = styled.span`
  margin-left: 0.5rem;
`;

export const EmptyList = styled.div`
  margin: 3rem 0;
  text-align: center;
  font-style: italic;
`;

export const StyledCell = styled.td(
  ({ theme }) => css`
    ${baseCellStyles(theme)}
    font-weight: 100;

    &:nth-child(4) {
      white-space: nowrap;
    }
  `,
);

export const Wrap = styled.div`
  display: flex;
  align-items: center;
`;
