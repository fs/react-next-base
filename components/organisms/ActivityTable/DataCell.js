import React from 'react';
import styled, { css } from 'styled-components';

import baseCellStyles from './baseCellStyles';

const StyledCell = styled.td(
  ({ theme }) => css`
    ${baseCellStyles(theme)}
    font-weight: 100;

    &:nth-child(4) {
      white-space: nowrap;
    }
  `,
);

const Wrap = styled.div`
  display: flex;
  align-items: center;
`;

const DataCell = ({ children }) => {
  return (
    <StyledCell>
      <Wrap>{children}</Wrap>
    </StyledCell>
  );
};

export default DataCell;
