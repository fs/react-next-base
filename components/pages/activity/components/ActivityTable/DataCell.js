import React from 'react';

import { StyledCell, Wrap } from './styled';

const DataCell = ({ children }) => {
  return (
    <StyledCell>
      <Wrap>{children}</Wrap>
    </StyledCell>
  );
};

export default DataCell;
