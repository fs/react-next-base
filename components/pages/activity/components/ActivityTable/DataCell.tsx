import React, { ReactNode } from 'react';

import { StyledCell, Wrap } from './styled';

const DataCell = ({ children }: { children: ReactNode }) => {
  return (
    <StyledCell>
      <Wrap>{children}</Wrap>
    </StyledCell>
  );
};

export default DataCell;
