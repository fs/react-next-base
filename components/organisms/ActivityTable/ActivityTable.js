import React from 'react';
import styled, { css } from 'styled-components';

import ProfileImage from 'components/atoms/ProfileImage';

import baseCellStyles from './baseCellStyles';
import DataCell from './DataCell';

const StyledTable = styled.table(
  ({ theme: { colors, up, breakpoints } }) =>
    css`
      position: relative;
      border-spacing: 0;
      display: block;
      overflow-x: auto;
      text-align: left;
      width: 100%;
      color: ${colors.darkGrey};

      ${up(breakpoints.lg)} {
        display: table;
        overflow-x: initial;
      }
    `,
);

const ColorLabel = styled.td(
  ({ color }) => css`
    min-width: 5px;
    background-color: ${color};
  `,
);

const HeaderCell = styled.th(({ theme }) => baseCellStyles(theme));

const UserInfo = styled.span`
  margin-left: 0.5rem;
`;

const EmptyList = styled.div`
  margin: 3rem 0;
  text-align: center;
  font-style: italic;
`;

const ActivityTable = ({ data }) => {
  const columnNames = ['Title', 'Description', 'Date', 'User'];

  return (
    <>
      {data.length > 0 ? (
        <StyledTable data-testid="test-activity-table">
          <thead>
            <tr>
              {columnNames.map((name, id) => (
                <HeaderCell key={name} colSpan={!id ? '2' : '1'}>
                  {name}
                </HeaderCell>
              ))}
            </tr>
          </thead>
          <tbody>
            {data.map(({ id, title, description, date, color, name, email, avatarUrl }) => (
              <tr key={id}>
                <ColorLabel color={color} />
                <DataCell>{title}</DataCell>
                <DataCell>{description}</DataCell>
                <DataCell>{date.toString()}</DataCell>
                <DataCell>
                  <ProfileImage avatar={avatarUrl} />
                  <UserInfo>
                    {name} ({email})
                  </UserInfo>
                </DataCell>
              </tr>
            ))}
          </tbody>
        </StyledTable>
      ) : (
        <EmptyList data-testid="test-activity-table-empty">No records found</EmptyList>
      )}
    </>
  );
};

export default ActivityTable;
