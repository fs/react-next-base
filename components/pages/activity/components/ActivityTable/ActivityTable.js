import React from 'react';

import ProfileImage from 'components/shared/atoms/ProfileImage';

import DataCell from './DataCell';

import { StyledTable, ColorLabel, HeaderCell, UserInfo, EmptyList } from './styled';

const ActivityTable = ({ data }) => {
  const columnNames = ['Title', 'Description', 'Date', 'User'];

  return (
    <>
      {data.length > 0 ? (
        <StyledTable data-testid="activity-table" data-cy="activity-table">
          <thead>
            <tr>
              {columnNames.map((name, id) => (
                <HeaderCell key={name} colSpan={!id ? '2' : '1'}>
                  {name}
                </HeaderCell>
              ))}
            </tr>
          </thead>
          <tbody>
            {data.map(({ id, title, description, date, color, name, email, avatarUrl }) => {
              return (
                <tr key={id} data-cy="activity-row" data-id={id}>
                  <ColorLabel color={color} />
                  <DataCell>{title}</DataCell>
                  <DataCell>{description}</DataCell>
                  <DataCell>{date.toString()}</DataCell>
                  <DataCell>
                    <ProfileImage avatar={avatarUrl} />
                    <UserInfo>
                      {name} ({email})
                    </UserInfo>
                  </DataCell>
                </tr>
              );
            })}
          </tbody>
        </StyledTable>
      ) : (
        <EmptyList data-testid="activity-table-empty">No records found</EmptyList>
      )}
    </>
  );
};

export default ActivityTable;
