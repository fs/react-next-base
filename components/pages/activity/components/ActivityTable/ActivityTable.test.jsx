import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';

import renderWithTheme from '__tests__/helpers/renderWithTheme';
import activitiesMock from '__tests__/mocks/activitiesMock';
import ActivityTable from 'components/pages/activity/components/ActivityTable';

describe('ActivityTable', () => {
  test('should render correctly', () => {
    // Arrange
    const mockData = activitiesMock;
    const mockTestId = 'activity-table';

    // Act
    render(renderWithTheme(<ActivityTable data={mockData} />));

    const table = screen.getByTestId(mockTestId);

    // Assert
    expect(table).toMatchSnapshot();
  });

  test('should render correctly if no data', () => {
    // Arrange
    const mockData = [];
    const mockTestId = 'activity-table-empty';

    // Act
    render(renderWithTheme(<ActivityTable data={mockData} />));

    const emptyText = screen.getByTestId(mockTestId);

    // Assert
    expect(emptyText).toBeInTheDocument();
  });
});
