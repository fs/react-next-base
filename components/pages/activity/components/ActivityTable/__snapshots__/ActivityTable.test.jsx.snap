// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityTable should render correctly 1`] = `
.c5 {
  position: relative;
  display: inline-block;
  width: 1.875rem;
  height: 1.875rem;
  font-size: 0;
  border-radius: 50%;
  overflow: hidden;
}

.c6 {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c0 {
  position: relative;
  border-spacing: 0;
  display: block;
  overflow-x: auto;
  text-align: left;
  width: 100%;
  color: #606c76;
}

.c2 {
  min-width: 5px;
  background-color: #2979ff;
}

.c8 {
  min-width: 5px;
  background-color: #00e676;
}

.c9 {
  min-width: 5px;
  background-color: #f50057;
}

.c1 {
  border-bottom: 1px solid #e1e1e1;
  padding: 0.8rem 0.6rem;
  font-size: 0.8rem;
  line-height: 1.6;
}

.c1:first-child {
  padding-left: 0;
}

.c7 {
  margin-left: 0.5rem;
}

.c3 {
  border-bottom: 1px solid #e1e1e1;
  padding: 0.8rem 0.6rem;
  font-size: 0.8rem;
  line-height: 1.6;
  font-weight: 100;
}

.c3:first-child {
  padding-left: 0;
}

.c3:nth-child(4) {
  white-space: nowrap;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:992.02px) {
  .c0 {
    display: table;
    overflow-x: initial;
  }
}

@media (min-width:992.02px) {
  .c1 {
    padding: 1.2rem 1rem;
    font-size: 0.9rem;
  }
}

@media (min-width:992.02px) {
  .c3 {
    padding: 1.2rem 1rem;
    font-size: 0.9rem;
  }
}

<table
  class="c0"
  data-cy="activity-table"
  data-testid="activity-table"
>
  <thead>
    <tr>
      <th
        class="c1"
        colspan="2"
      >
        Title
      </th>
      <th
        class="c1"
        colspan="1"
      >
        Description
      </th>
      <th
        class="c1"
        colspan="1"
      >
        Date
      </th>
      <th
        class="c1"
        colspan="1"
      >
        User
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      data-cy="activity-row"
      data-id="1"
    >
      <td
        class="c2"
        color="#2979ff"
      />
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          User Updated
        </div>
      </td>
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          User updated with the next attributes: First name - 111test1, Last name - еуые, Email - tes1t@test.com
        </div>
      </td>
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          04.08.2020, 13:38:51
        </div>
      </td>
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <img
              alt="Avatar"
              class="c6"
              src="/images/avatar-placeholder.png"
            />
          </div>
          <span
            class="c7"
          >
            111test1 еуые
             (
            tes1t@test.com
            )
          </span>
        </div>
      </td>
    </tr>
    <tr
      data-cy="activity-row"
      data-id="6"
    >
      <td
        class="c8"
        color="#00e676"
      />
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          User Logged In
        </div>
      </td>
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          User logged in with the next attributes: First name - , Last name - 
        </div>
      </td>
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          04.08.2020, 15:01:32
        </div>
      </td>
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <img
              alt="Avatar"
              class="c6"
              src="/images/avatar-placeholder.png"
            />
          </div>
          <span
            class="c7"
          >
            Nikita Polyakov
             (
            nikita.polyakov@flatstack.com
            )
          </span>
        </div>
      </td>
    </tr>
    <tr
      data-cy="activity-row"
      data-id="28"
    >
      <td
        class="c9"
        color="#f50057"
      />
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          User Registered
        </div>
      </td>
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          New user registered with the next attributes: First name - Test, Last name - Test↵
        </div>
      </td>
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          04.08.2020, 15:26:08
        </div>
      </td>
      <td
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <img
              alt="Avatar"
              class="c6"
              src="/images/avatar-placeholder.png"
            />
          </div>
          <span
            class="c7"
          >
            Test Test
             (
            test123@example.com
            )
          </span>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
